src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/cache/SimpleResourceCache.java

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) 2015 OpenPlans - www.openplans.org. All rights reserved.
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.jdbcstore.cache;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resource.Type;
+
+/**
+ * Basic implementation for Resource Cache.
+ * 
+ * @author Kevin Smith, Boundless
+ * @author Niels Charlier
+ *
+ */
+public class SimpleResourceCache implements ResourceCache {
+    File base;
+    boolean cacheChildren = true;
+    
+    public SimpleResourceCache() {        
+    }
+        
+    public SimpleResourceCache(File base) {
+        this.base=base;
+    }    
+    
+    public File getBase() {
+        return base;
+    }
+
+    public void setBase(File base) {
+        this.base = base;
+    }
+    
+    public boolean isCacheChildren() {
+        return cacheChildren;
+    }
+
+    public void setCacheChildren(boolean cacheChildren) {
+        this.cacheChildren = cacheChildren;
+    }
+
+    void cacheData(Resource res, File file) throws IOException {
+        assert res.getType()==Type.RESOURCE;
+        OutputStream out = new FileOutputStream(file);
```

Comment:
Could this be a single try-with-resources block with two resources and a single statement in the body?